File: _src/05/Chapter5.CSharp/Maps/AddressMap.cs
using System.Data.Entity.ModelConfiguration;
using Chapter5.CSharp.Models;

namespace Chapter5.CSharp.Maps
{
public class AddressMap : ComplexTypeConfiguration<Address>
{
    public AddressMap()
    {
        Property(p => p.Street)
            .HasMaxLength(40)
            .IsRequired()
            .HasColumnName("Street");
        Property(p => p.City)
            .HasMaxLength(30)
            .IsRequired()
            .HasColumnName("City");
        Property(p => p.State)
            .HasMaxLength(2)
            .IsRequired()
            .HasColumnName("State");
        Property(p => p.Zip)
            .HasMaxLength(5)
            .IsRequired()
            .HasColumnName("Zip");
    }
}
}

File: _src/05/Chapter5.VB/Maps/AddressMap.vb
Imports System.Data.Entity.ModelConfiguration
Imports Chapter5.VB.Models

Namespace Maps

Public Class AddressMap
    Inherits ComplexTypeConfiguration(Of Address)
    Public Sub New()
        Me.Property(Function(p) p.Street) _ 
            .HasMaxLength(40) _ 
            .IsRequired() _ 
            .HasColumnName("Street")
        Me.Property(Function(p) p.City).HasMaxLength(30).IsRequired().HasColumnName("City")
        Me.Property(Function(p) p.State).HasMaxLength(2).IsRequired().HasColumnName("State")
        Me.Property(Function(p) p.Zip).HasMaxLength(5).IsRequired().HasColumnName("Zip")
    End Sub
End Class
End NameSpace